src/runtime/Types/ClassBase.cs

```diff
@@ -276,42 +275,46 @@ public static nint tp_hash(BorrowedReference ob)
         /// </summary>
         public static NewReference tp_str(BorrowedReference ob)
         {
-            var co = GetManagedObject(ob) as CLRObject;
-            if (co == null)
-            {
-                return Exceptions.RaiseTypeError("invalid object");
-            }
-            try
-            {
-                return Runtime.PyString_FromString(co.inst.ToString());
-            }
-            catch (Exception e)
+            if (GetManagedObject(ob) is CLRObject co)
             {
-                if (e.InnerException != null)
+                try
                 {
-                    e = e.InnerException;
+                    return Runtime.PyString_FromString(co.inst.ToString());
+                }
+                catch (Exception e)
+                {
+                    if (e.InnerException != null)
+                    {
+                        e = e.InnerException;
+                    }
+                    Exceptions.SetError(e);
+                    return default;
                 }
-                Exceptions.SetError(e);
-                return default;
             }
+            return Exceptions.RaiseTypeError("invalid object");
```

Comment:
I think keeping input parameter validation returns before the main body is a good practice.